docs: include `CONTRIBUTING.md` for dart_frog (#648)
VeryGoodOpenSource/dart_frog: packages/dart_frog_cli/CONTRIBUTING.md

# 🦄 Contributing to Dart Frog CLI

First of all, thank you for taking the time to contribute! 🎉👍 Before you do, please carefully read this guide.

## Understanding Dart Frog CLI

> The Dart Frog CLI is a [`CommandRunner`](https://pub.dev/documentation/args/latest/command_runner/CommandRunner-class.html) that exposes four visible commands:

Reviewer: ```suggestion
The Dart Frog CLI is a [`CommandRunner`](https://pub.dev/documentation/args/latest/command_runner/CommandRunner-class.html) that exposes some commands such as:
```

Reviewer 2: 👍

> - build: Creates a production build by using the [`dart_frog_prod_server`](../../bricks/dart_frog_prod_server/) mason template to generate files.
> - create: Creates a new Dart Frog app by using the [`dart_frog_new`](../../bricks/dart_frog_new/) mason template to generate the files.
> - dev: Runs a local development server at a given port and listens to file changes generating files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template to generate file

Reviewer 2: ```suggestion
- `build`: Creates a production build by using the [`dart_frog_prod_server`](../../bricks/dart_frog_prod_server/) mason template to generate files.
- `create`: Creates a new Dart Frog app by using the [`dart_frog_new`](../../bricks/dart_frog_new/) mason template to generate the files.
- `dev`: Runs a local development server at a given port and listens to file changes generating files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template to generate file
```

for `dev`: should it be "...and listens to file changes using" or "...and listens to generating files using"; seems like it shouldn't be both "...and listens to file changes generating files using"

> - dev: Runs a local development server at a given port and listens to file changes generating files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template to generate file

Author: ```suggestion
- dev: Runs a local development server at a given port and listens to file changes, generating new server files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template on updates
```

> - `dev`: Runs a local development server at a given port and listens to file changes generating files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template to generate file

Author: ```suggestion
- `dev`: Runs a local development server at a given port and listens to file changes, generating new files using the [`dart_frog_dev_server`](../../bricks/dart_frog_dev_server/) mason template on updates.
```

> - update: Updates the Dart Frog CLI's if possible.

Author: Do we want to include this information here or should we point to the actual command classes? Any  other suggestions?

Reviewer: It can totally be removed. The docs already talk about the sub-commands available and are more easily updatable

> - update: Updates the Dart Frog CLI's if possible.

Reviewer 2: ```suggestion
- `update`: Updates the Dart Frog CLI's if possible.
```

2. Install [Very Good CLI](https://github.com/VeryGoodOpenSource/very_good_cli):

```sh
# 💻 Install Very Good CLI globally:
dart pub global activate very_good_cli
```

3. Install all Dart Frog CLI's dependencies:

```sh
# 📂 Get project dependencies recursively with Dart Frog CLI's
very_good packages get -r --ignore="bricks/**"
```

3. Run all Dart Frog CLI's tests:

```sh
# 🧪 Run Dart Frog CLI's unit tests (from packages/dart_frog_cli)
dart test

# 🧪💻 Run Dart Frog CLI's end to end test (from packages/dart_frog_cli/e2e)
dart test
```

If some tests do not pass out of the box, please submit an [issue](https://github.com/VeryGoodOpenSource/dart_frog/issues/new/choose).

4. Install your own version of Dart Frog CLI in your local environment:

```sh
# 🚀 Activate your own local version of Dart Frog CLI (from packages/dart_frog_cli)
dart pub global activate --source path .
```

> 5. If you are modifying any templates, make sure to bundle them before activating:

Reviewer: Should we specify where those templates are located at?

Reviewer 2: If you think we should -- it can't hurt!
